### ISSUE TITLE
Catalog Page Empty if "Allow All Products Per Page" is set to "Yes"

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#35900]
### Preconditions and environment

- Magento 2.4.5

### Steps to reproduce

Deploy 2.4.5 vanilla
Create Test Category
Create Test Product in test category
Browse to test category page on frontend and ensure product is visible.
Set Allow All Products Per Page setting (Config -> Catalog -> Catalog) to Yes
![image](https://user-images.githubusercontent.com/30697781/183713911-36281be3-74d8-41fb-8a8a-4c62cf887bb1.png)

Clear Cache
Browse to test category page on frontend and ensure product is visible.

### Expected result

Product is visisble
![image](https://user-images.githubusercontent.com/30697781/183714059-b4cb4b10-33a5-4903-a5be-b50664872884.png)

### Actual result

Product is not visible
![image](https://user-images.githubusercontent.com/30697781/183713857-54495c03-8930-4b7d-8d3e-7b9209481c7d.png)

### Release note

The Category Page is no longer empty when "Allow All Products Per Page" is set to "Yes".

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by gwharton** [comment]
@magento give me 2.4-develop instance

**Comment by gwharton** [comment]
Confirmed on magento test instance.
![image](https://user-images.githubusercontent.com/30697781/183715886-d4a106c7-99e5-4045-b541-89348ab64cdb.png)

**Comment by Kannakiraj123** [comment]
@magento give me 2.4-develop instance

**Comment by thlassche** [comment]
Workaround:

Override `limiter.phtml` in your theme:

`app/design/frontend/Your/Theme/Magento_Catalog/templates/product/list/toolbar/limiter.phtml`

```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */
?>
<?php
/**
 * Product list toolbar
 *
 * @var \Magento\Catalog\Block\Product\ProductList\Toolbar $block
 * @var \Magento\Framework\Locale\LocaleFormatter $localeFormatter
 */
?>
<div class="field limiter">
    <label class="label" for="limiter">
        <span><?= $block->escapeHtml(__('Show')) ?></span>
    </label>
    <div class="control">
        <select id="limiter" data-role="limiter" class="limiter-options">
            <?php foreach ($block->getAvailableLimit() as $_key => $_limit):?>
                <option value="<?= $block->escapeHtmlAttr($_key) ?>"
                    <?php if ($block->isLimitCurrent($_key)):?>
                        selected="selected"
                    <?php endif ?>>
                    <?= $block->escapeHtml($_limit) ?>
                </option>
            <?php endforeach; ?>
        </select>
    </div>
    <span class="limiter-text"><?= $block->escapeHtml(__('per page')) ?></span>
</div>
```

**Comment by stephansteiner** [comment]
Having the same issue, apparently the $localeFormatter (which is new in 2.4.5) is not passed to the product/list/toolbar/limiter.phtml template or any other template like product/list/toolbar/amount.phtml or product/price/tier_prices.phtml at all.

Upgrading a webshop to 2.4.5 gives me the error:
Warning: Undefined variable $localeFormatter in /var/www/html/vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/limiter.phtml on line 26

**Comment by gwharton** [comment]
Whilst my error was on Line 26, it was because it was trying to parse the phrase "All" as an int. It was expecting all entries of the limiter to be an integer number which it isnt when the setting is set.

I don't seem to be experiencing the $localeFormatter issue.

**Comment by thlassche** [comment]
@gwharton Same in my case.

**Comment by hostep** [comment]
Closed without a comment?
But I see one commit mentioning `AC-6384`:  https://github.com/magento/magento2/commit/bb55549cd3016987663272e7ffe3f452c8d6e40d, so that's probably the fix

**Comment by gwharton** [comment]
PR is here https://github.com/magento/magento2/pull/35987

**Comment by hostep** [comment]
Which got closed with a comment:
> Resolved internally

**Comment by engcom-Hotel** [comment]
Hello,

As I can see this issue got fixed in the scope of the internal Jira ticket AC-6384 by the internal team
Related commits: https://github.com/magento/magento2/search?q=AC-6384&type=commits

Based on the Jira ticket, the target version is 2.4.6.

Thanks

**Comment by dragonfly4** [comment]
I had the same issue. Can confirm that PR #35987 solves the pagination loading issue.

**Comment by hostep** [comment]
@dragonfly4, the official fix (which is a better one), is over here: https://github.com/magento/magento2/commit/bb55549cd3016987663272e7ffe3f452c8d6e40d

**Comment by rarog** [comment]
This was obviously patched before 2.4.5-p1 was published but was omitted from the release. If it's due to be published in 2.4.6 like mentioned by @engcom-Hotel this would mean that there won't be a fix before [2.4.5-p2/2.4.6 and the open source release in March 2023](https://experienceleague.adobe.com/docs/commerce-operations/release/schedule.html). That's quite a long time for a quick fix.

**Comment by andreakommerce** [comment]
> Having the same issue, apparently the $localeFormatter (which is new in 2.4.5) is not passed to the product/list/toolbar/limiter.phtml template or any other template like product/list/toolbar/amount.phtml or product/price/tier_prices.phtml at all.
> 
> Upgrading a webshop to 2.4.5 gives me the error: Warning: Undefined variable $localeFormatter in /var/www/html/vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/limiter.phtml on line 26

@stephansteiner have you find a solution for this issue?
